mobile/lib/auth/email_auth_widget.dart

```diff
@@ -608,12 +489,118 @@ class EmailAuthWidgetState<T extends EmailAuthWidget> extends State<T> {
+
+    if (!_emailFormKey.currentState!.validate()) {
+      return;
+    }
+
+    FocusScope.of(context).requestFocus(FocusNode());
+    Future.delayed(const Duration(milliseconds: 400), () {
+      setState(() {
+        _showAuthOptions = true;
+      });
+    });
+
+    setState(() {
+      _nextBtnColor = Config.appColorDisabled;
+    });
+    loadingScreen(loadingContext);
+
+    if (!widget.isLogin) {
+      var emailExists = await _appService.doesUserExist('', _emailAddress);
+
+      if (emailExists) {
+        setState(() {
+          _nextBtnColor = Config.appColorBlue;
+        });
+        Navigator.pop(loadingContext);
+        await showSnackBar(
+            context,
+            'You already have an '
+            'account with this email address');
+        await Navigator.pushAndRemoveUntil(
+            context,
+            PageRouteBuilder(
+              pageBuilder: (context, animation, secondaryAnimation) =>
+                  EmailLoginWidget(
+                emailAddress: _emailAddress,
+                enableBackButton: false,
+              ),
+              transitionsBuilder:
+                  (context, animation, secondaryAnimation, child) {
+                return FadeTransition(
+                  opacity: animation.drive(Tween<double>(begin: 0, end: 1)),
+                  child: child,
+                );
+              },
+            ),
+            (r) => false);
+
+        return;
+      }
+    }
+
+    var emailSignupResponse = await _appService.apiClient
+        .requestEmailVerificationCode(_emailAddress, false);
+
+    Navigator.pop(loadingContext);
+
+    if (emailSignupResponse == null) {
+      await showSnackBar(context, 'email signup verification failed');
+      setState(() {
+        _nextBtnColor = Config.appColorBlue;
+      });
+      return;
+    }
+
+    setState(() {
+      _emailVerificationLink = emailSignupResponse.loginLink;
+      _emailToken = emailSignupResponse.token;
+      _verifyCode = true;
+      _codeSent = false;
+    });
+
+    _startCodeSentCountDown();
+  }
+
+  Future<void> _resendVerificationCode() async {
+    var connected = await _appService.isConnected();
+    if (!connected) {
+      await showSnackBar(context, Config.connectionErrorMessage);
+      return;
+    }
+
+    loadingScreen(loadingContext);
+
+    var emailSignupResponse = await _appService.apiClient
+        .requestEmailVerificationCode(_emailAddress, false);
+
+    Navigator.pop(loadingContext);
+
+    if (emailSignupResponse == null) {
+      await showSnackBar(context, 'Email signup verification failed');
+      return;
+    }
+
+    setState(() {
+      _emailVerificationLink = emailSignupResponse.loginLink;
+      _emailToken = emailSignupResponse.token;
+    });
+    _startCodeSentCountDown();
+  }
+
   void _startCodeSentCountDown() {
     setState(() {
       _codeSentCountDown = 5;
     });
     Timer.periodic(
-      const Duration(seconds: 1),
+      const Duration(milliseconds: 1200),
```

Comment:
Thanks @NoahNsimbe , are we able to use env variables for this duration figure?

Also somehow related, the duration for the flash screens on subsequent app visits, can that also be reduced or removed. I am not sure about the significance of seeing the flash screen on after the first visit of the application.

Reply:
This is not for splash screen. Its displays a countdown after you have entered your phone number. I shall screenshoots